lib/fastlane/plugin/wpmreleasetoolkit/models/build_number.rb

```diff
@@ -0,0 +1,38 @@
+require 'xcodeproj'
+
+module Fastlane
+  module Model
+    class BuildNumber
+      attr_accessor :file_path
+
+      def initialize(file_path)
+        @file_path = file_path
+      end
+
+      def get_build_number
+        read_build_number_from_xcconfig.to_s
+      end
+
+      def read_build_number_from_xcconfig
+        config = Xcodeproj::Config.new(@file_path)
+
+        UI.user_error!('The .xcconfig file doesn\'t have a BUILD_NUMBER configured') if config.attributes['BUILD_NUMBER'].nil?
+
+        config.attributes['BUILD_NUMBER']
+      end
+
+      def write_build_number_to_xcconfig(build_number)
+        if File.exist?(@file_path)
+          Action.sh("sed -i '' \"$(awk '/^BUILD_NUMBER/{ print NR; exit }' \"#{@file_path}\")s/=.*/=#{build_number}/\" \"#{@file_path}\"")
```

Comment:
`awk` and `sed` behave very differently on macOS and Linux, and as a rule we should avoid calling out to `sh` unless we're puppeteering some other CLI tool we can't use any other way. 

I'd strongly suggest using the `xcodeproj` gem for parsing and writing to `xcconfig` files.